ee/codegen/src/__test__/nodes/subworkflow-deployment-node.test.ts

```diff
@@ -29,8 +29,8 @@ describe("SubworkflowDeploymentNode", () => {
       ).mockResolvedValue({
         name: "test-deployment",
         outputVariables: [
-          { id: "1", key: "output-1", type: "STRING" },
-          { id: "2", key: "output-2", type: "NUMBER" },
+          { id: "1", key: "output-1", type: "STRING", required: true },
+          { id: "2", key: "output-2", type: "NUMBER", required: true },
```

Comment:
I led you astray here, check out this thread: https://vellum-ai.slack.com/archives/C06P13ABK0W/p1739308671509959?thread_ts=1739302036.746689&cid=C06P13ABK0W

Basically, the `undefined` behavior is dependent on the context. So for workflow inputs, undefined means required: false, and for workflow outputs, undefined means required: true. There are legacy reasons driving this confusion